x-pack/plugins/maps/public/connected_components/timeslider/timeslider.tsx

```diff
@@ -164,9 +199,87 @@ class KeyedTimeslider extends Component<Props, State> {
     });
   }
 
+  _togglePopover = () => {
+    this.setState((prevState) => ({
+      isPopoverOpen: !prevState.isPopoverOpen,
+    }));
+  };
+
+  _closePopover = () => {
+    this.setState({
+      isPopoverOpen: false,
+    });
+  };
+
+  _renderFilteredPeriods = () => {
+    const filteredPeriods = getTimeRanges(this.defaultRange.timeRangeBounds).map(
+      (range: { label: string; ms: number }, index: number) => {
+        return (
+          <EuiContextMenuItem
+            key={index}
+            onClick={() => {
+              this._updateTimeRange(range);
+            }}
+          >
+            {range.label}
+          </EuiContextMenuItem>
+        );
+      }
+    );
+
+    filteredPeriods?.unshift(
+      <EuiContextMenuItem
+        key="default"
+        onClick={() => {
+          this._updateTimeRange({
+            label: i18n.translate('xpack.maps.timeslider.autoPeriod', {
+              defaultMessage: 'Auto',
+            }),
+            ms: 0,
+            default: true,
+          });
+        }}
+      >
+        {i18n.translate('xpack.maps.timeslider.autoPeriod', {
+          defaultMessage: 'Auto',
+        })}
+      </EuiContextMenuItem>
+    );
+
+    return filteredPeriods;
+  };
+
+  _updateTimeRange = (data: { label: string; ms: number; default?: boolean }) => {
```

Comment:
_updateTimeRange should also call `_onChange` to snap the timeslice to the newly selected interval.

Reply:
Done